packages/components/src/InputNumber/InputNumber.tsx

```diff
@@ -57,11 +58,29 @@ function InputNumberInternal(
       onChange={handleChange}
       validations={{
         ...props.validations,
-        validate: getOverLimitMessage,
+        validate: customValidators(props.validations?.validate),
       }}
     />
   );
 
+  function customValidators(
+    validators?: RegisterOptions["validate"],
+  ): RegisterOptions["validate"] {
+    if (validators == null) {
+      return getOverLimitMessage;
+    } else if (typeof validators === "function") {
+      return {
+        customValidation: validators,
+        getOverLimitMessage,
+      };
```

Comment:
This seems to not fire `getOverLimitMessage` anymore. Same goes for the one below

Also, good to have a test case for when you have a custom validator the min/max should still work.

Reply:
Turns out, if one of your `validate` functions returns a `false` it'll prevent the other validations from happening.

I think the only change here is the test to make sure the internal validations still runs and document the validation quirkiness